sysutils/edid-decode/files/patch-destdir-variable-fix.diff

```diff
@@ -0,0 +1,13 @@
+--- Makefile.orig	2022-05-17 21:54:35.000000000 -0400
```

Comment:
This patch is not working for MacPorts installations in locations other than `/opt/local`.

Perhaps something like this:
```
--- ./Makefile	2023-06-07 21:20:51.000000000 -0400
+++ ./Makefile	2023-06-07 21:51:31.000000000 -0400
@@ -1,17 +1,16 @@
+
 ifeq ($(OS),Windows_NT)
-	bindir ?= /usr/bin
-	mandir ?= /usr/share/man
+        PREFIX ?= /usr
 else
 	UNAME_S := $(shell uname -s)
 	ifeq ($(UNAME_S),Darwin)
-		bindir ?= /usr/local/sbin
-		mandir ?= /usr/local/share/man
-	else
-		bindir ?= /usr/bin
-		mandir ?= /usr/share/man
+		PREFIX ?= /opt/local
 	endif
 endif

+bindir ?= $(PREFIX)/bin
+mandir ?= $(PREFIX)/share/man
+
 EMXX ?= em++

 SOURCES = $(wildcard *.cpp)
```

Reply:
The reason why Herby's proposal would work in other prefixes, even though it still says /opt/local, is that the makefile portgroup will set the PREFIX environment variable to the correct prefix.

However, ideally, patches such as these should be written with an eye toward submitting them to the developers for inclusion upstream, so that we don't need to maintain the patches forever, and this revised patch as written would not be suitable for upstreaming because it deletes the non-Windows-NT non-Darwin code path so PREFIX would be unset on those systems, and also because it defaults the Darwin installation prefix to /opt/local; we would not want non-MacPorts Darwin users to install this software there. /usr/local seems to me like a reasonable default prefix for all Unix-like systems, but they may prefer to continue to use /usr for non-Darwin Unix-like systems for whatever reason.

Reply:
The Makefile also mistakenly uses `CFLAGS` instead of `CXXFLAGS`.

Reply:
I attempted to make a more general-purpose patch (d73a890), which as far as I can tell preserves the original behavior across platforms but allows MacPorts to override `PREFIX`:

```Makefile
ifeq ($(OS),Windows_NT)
        PREFIX ?= /usr
else
        UNAME_S := $(shell uname -s)
        ifeq ($(UNAME_S),Darwin)
                PREFIX ?= /usr/local
        else
                PREFIX ?= /usr
        endif
endif

bindir ?= $(PREFIX)/bin
mandir ?= $(PREFIX)/share/man
```

The patch also swaps `CFLAGS` for `CXXFLAGS`.

Reply:
I just realized the same result can be accomplished without a patch by simply setting the variables in `destroot.args`:

```
destroot.args-append bindir=${prefix}/bin \
                     mandir=${prefix}/share/man
```